yt_dlp/extractor/taptap.py

```diff
@@ -0,0 +1,209 @@
+            'uploader': '乌酱',
+        },
+        'playlist_count': 1,
+        'playlist': [{
+            'info_dict': {
+                'id': '2202584',
+                'ext': 'mp4',
+                'title': '《崩坏3》开放世界「后崩坏书」新篇章 于淹没之地仰视辰星',
+                'description': 'md5:cf66f7819d413641b8b28c8543f4ecda',
+                'duration': 66,
+                'timestamp': 1633453402,
+                'upload_date': '20211005',
+                'uploader': '乌酱',
+                'thumbnail': r're:^https?://.*\.(png|jpg)',
+            }
+        }]
+    }, {
+        'url': 'https://www.taptap.cn/moment/521630629209573493',
+        'info_dict': {
+            'id': '521630629209573493',
+            'title': '《崩坏：星穹铁道》黄泉角色PV——「你的颜色」',
+            'description': 'md5:2c81245da864428c904d53ae4ad2182b',
+            'timestamp': 1711425600,
+            'upload_date': '20240326',
+            'uploader': '崩坏：星穹铁道',
+        },
+        'playlist_count': 1,
+        'playlist': [{
+            'info_dict': {
+                'id': '4006511',
+                'ext': 'mp4',
+                'title': '《崩坏：星穹铁道》黄泉角色PV——「你的颜色」',
+                'description': 'md5:2c81245da864428c904d53ae4ad2182b',
+                'duration': 173,
+                'timestamp': 1711425600,
+                'upload_date': '20240326',
+                'uploader': '崩坏：星穹铁道',
+                'thumbnail': r're:^https?://.*\.(png|jpg)',
+            }
+        }]
+    }]
+
+
+class TapTapAppIE(TapTapBaseIE):
+    _VALID_URL = r'https?://www\.taptap\.cn/app/(?P<id>\d+)'
+    _INFO_API = 'https://www.taptap.cn/webapiv2/app/v4/detail'
+    _ID_PATH = (('app_videos', 'videos'), ..., 'video_id')
+    _META_PATH = {
+        'title': ('title', {str}),
+        'description': ('description', 'text', {str}, {clean_html}),
+    }
+    _TESTS = [{
+        'url': 'https://www.taptap.cn/app/168332',
+        'info_dict': {
+            'id': '168332',
+            'title': '原神',
+            'description': 'md5:e345f39a5fea5de2a46923f70d5f76ab',
+        },
+        'playlist_count': 2,
+        'playlist': [{
+            'info_dict': {
+                'id': '4058443',
+                'ext': 'mp4',
+                'title': '原神',
+                'description': 'md5:e345f39a5fea5de2a46923f70d5f76ab',
+                'duration': 26,
+                'thumbnail': r're:^https?://.*\.(png|jpg)',
+            }
+        }, {
+            'info_dict': {
+                'id': '4058462',
+                'ext': 'mp4',
+                'title': '原神',
+                'description': 'md5:e345f39a5fea5de2a46923f70d5f76ab',
+                'duration': 295,
+                'thumbnail': r're:^https?://.*\.(png|jpg)',
+            }
+        }]
+    }]
+
+
+class TapTapAppIntlIE(TapTapAppIE, TapTapIntlBase):
+    _VALID_URL = r'https?://www\.taptap\.io/app/(?P<id>\d+)'
+    _INFO_API = 'https://www.taptap.io/webapiv2/i/app/v5/detail'
+    _DATA_PATH = ('data', 'app')
+
```

Comment:
I think this little bit of duplication is fine. Let's also add a test case.
```suggestion
    _ID_PATH = (('app_videos', 'videos'), ..., 'video_id')
    _META_PATH = {
        'title': ('title', {str}),
        'description': ('description', 'text', {str}, {clean_html}),
    }
    _TESTS = [{
        'url': 'https://www.taptap.io/app/233287',
        'info_dict': {
            'id': '233287',
            'title': '《虹彩六號 M》',
            'description': 'md5:418285f9c15347fc3cf3e3a3c649f182',
        },
        'playlist_count': 1,
        'playlist': [{
            'info_dict': {
                'id': '2149708997',
                'ext': 'mp4',
                'title': '《虹彩六號 M》',
                'description': 'md5:418285f9c15347fc3cf3e3a3c649f182',
                'duration': 78,
                'thumbnail': r're:^https?://.*\.(png|jpg)',
            }
        }],
        'params': {'skip_download': 'm3u8'},
    }]

```